test/lib/test-xhr-call.js

```diff
@@ -1,58 +1,61 @@
 module.exports = testXHRCall;
 
 var sinon = require('sinon');
-var AlgoliaSearch = require('algoliasearch');
 var url = require('url');
 
+var AlgoliaSearch = require('algoliasearch');
+
 function testXHRCall(opts) {
   var t = opts.test;
 
   var xhrMock = sinon.useFakeXMLHttpRequest();
   var xhr;
-
   xhrMock.onCreate = function(newXhr) {
     xhr = newXhr;
   };
 
   var client = new AlgoliaSearch(opts.applicationID, opts.searchOnlyAPIKey);
   var object;
-
   if (opts.object === 'index') {
     object = client.initIndex(opts.indexName);
   } else {
     object = client;
   }
 
-  opts.call.args.push(function(success, content) {
-    t.deepEqual(
-      url.parse(xhr.url),
-      url.parse(url.format(opts.call.expectedXhr.url, true)),
-      'Request url matches'
-    );
-
-    t.deepEqual(
-      JSON.parse(xhr.requestBody),
-      opts.call.expectedXhr.body,
-      'Request body matches'
-    );
-
-    t.deepEqual(
-      xhr.requestHeaders,
-      opts.call.expectedXhr.headers,
-      'Request headers matches'
-    );
-
-    console.log(content)
-  });
-
   object[opts.methodName].apply(object, opts.call.args);
 
   xhr.respond(
-    opts.call.response.statusCode,
-    opts.call.response.headers,
-    JSON.stringify(opts.call.response.body)
+    opts.call.fakeResponse.statusCode,
+    opts.call.fakeResponse.headers,
+    JSON.stringify(opts.call.fakeResponse.body)
+  );
+
+  t.deepEqual(
+    url.parse(xhr.url),
+    url.parse(url.format(opts.call.expectedRequest.url, true)),
+    'Request url matches'
+  );
+
+  t.deepEqual(
+    JSON.parse(xhr.requestBody),
+    opts.call.expectedRequest.body,
+    'Request body matches'
+  );
+
+  t.deepEqual(
+    xhr.requestHeaders,
+    opts.call.expectedRequest.headers,
+    'Request headers matches'
   );
 
+
+  opts.call.args.forEach( function doTestCheck( p ){
```

Comment:
p stands for param, why not write `param`? We should try to avoid single letter arguments, what do you think?